src/metabase/db.clj

```diff
@@ -1,49 +1,47 @@
-  sub-namespaces:
+  "API namespace for the application database (app-db).
 
-  * [[metabase.db.connection]] - functions for getting the application database type (e.g. `:h2`) and a
-    [[clojure.java.jdbc]] spec for it; dynamic variable for rebinding it
+  It has a few different functions you might need:
+  - A connectible for the app-db [[app-db]] and [[data-source]]
+  - Information about the app-db: [[db-is-set-up?]], [[db-type]], [[quoting-style]]
+  - a few other random functions that have built up for different purposes.
 
-  * [[metabase.db.connection-pool-setup]] - functions for creating a connection pool for the application database
-
-  * [[metabase.db.custom-migrations]] - Clojure-land data migration definitions and functions for running them
-
-  * [[metabase.db.data-source]] - Implementations of [[javax.sql.DataSource]] for raw connection strings and
-    broken-out db details. See [[metabase.db.env/broken-out-details]] for more details about what 'broken-out details'
-    means.
-
-  * [[metabase.db.env]] - functions for getting application database connection information from environment variables
-
-  * [[metabase.db.jdbc-protocols]] - implementations of [[clojure.java.jdbc]] protocols for the Metabase application
-    database
-
-  * [[metabase.db.liquibase]] - high-level Clojure wrapper around relevant parts of the Liquibase API
-
-  * [[metabase.db.setup]] - code related to setting up the application DB -- verifying the connection and running
-    migrations -- and for setting it up as the default Toucan connection
-
-  * [[metabase.db.spec]] - util functions for creating JDBC specs for supported application DB types from connection
-    details maps
-
-  * [[metabase.db.util]] - general util functions for Toucan/HoneySQL queries against the application DB"
+  Other namespaces should only depend on this namespace and a sensible API for the app-db maintained."
   (:require
-   [clojure.core.async.impl.dispatch :as a.impl.dispatch]
    [metabase.config :as config]
    [metabase.db.connection :as mdb.connection]
+   [metabase.db.connection-pool-setup :as mdb.connection-pool-setup]
+   [metabase.db.env :as mdb.env]
    [metabase.db.setup :as mdb.setup]
-   [methodical.core :as methodical]
-   [potemkin :as p]
-   [toucan2.pipeline :as t2.pipeline]))
+   [metabase.db.spec :as mdb.spec]
+   [potemkin :as p]))
+
+(set! *warn-on-reflection* true)
 
 ;; TODO - determine if we *actually* need to import any of these
 ;;
 ;; These are mostly here as a convenience to avoid having to rework a bunch of existing code. It's better to use these
 ;; functions directly where applicable.
 (p/import-vars
  [mdb.connection
+  quoting-style
   db-type
-  quoting-style])
+  unique-identifier
+  data-source]
+
+ [mdb.connection-pool-setup
+  recent-activity?]
+
+ [mdb.env
+  db-file]
+
+ [mdb.setup
+  migrate!]
+
+ [mdb.spec
+  make-subname
+  spec]
+ )
```

Comment:
do we really have dangling parens for potemkin imports? Would've expected this to be on the previous line

Reply:
fixed